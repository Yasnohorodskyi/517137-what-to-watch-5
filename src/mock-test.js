import configureMockStore from "redux-mock-store";
import {createStore} from 'redux';
import rootReducer from 'store/reducers/root-reducer';
import {setPromoFilm} from "./store/action";

const mockStore = configureMockStore();
const store = createStore(rootReducer);
export const noop = () => {};
export const USER_DATA = {login: `my@mail.ru`, password: `kooliman`};
export const VISIBLE_MOVIES_COUNT = 16;
export const ID = 1;
export const RATING_STARS = [false, false, true, false, false];
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const CURRENT_GENRE = `Comedy`;
export const ALL_GENRES = `All genres`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {params: {id: 1}};
export const PARAMS = {id: 1};

export const FILMS = [
  {
    "id": 1,
    "name": `Bronson`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    "background_color": `#73B39A`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    "description": `A young man who was sentenced to seven years`,
    "rating": 3.6,
    "scores_count": 109661,
    "director": `Nicolas Winding Refn`,
    "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    "run_time": 92,
    "genre": `Action`,
    "released": 2008,
    "is_favorite": true
  },
  {
    "id": 2,
    "name": `Matrix`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
    "background_color": `#73B39A`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    "description": `A young man who was sentenced to seven years`,
    "rating": 8.6,
    "scores_count": 109661,
    "director": `Nicolas Winding Refn`,
    "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    "run_time": 92,
    "genre": `Action`,
    "released": 2008,
    "is_favorite": true
  },
];

export const FILM = {
  "id": 5,
  "name": `Pumping`,
  "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
  "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
  "background_color": `#795433`,
  "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  "preview_video_link": `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  "description": `A young man who was sentenced to seven years`,
  "rating": 8.6,
  "scores_count": 109661,
  "director": `Nicolas Winding Refn`,
  "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  "run_time": 92,
  "genre": `Action`,
  "released": 2008,
  "is_favorite": true
};


export const REVIEWS = [
  {
    id: 111,
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    author: `Christina`,
    message: `This film is an experience and i has already seen it 4 times`,
  },
  {
    id: 222,
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    author: `Christina`,
    message: `This film is an experience and i has already seen it 4 times`,
  }
];

export const MockComponent = (props) => {
  const {children} = props;

  return (
    <>
      {children}
    </>
  );
};

export const activeState = {
  reviewRating: ``,
  reviewText: ``,
  error: false,
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export const storeAuth = mockStore({
  USER: {
    userData: {
      data: {
        id: 1,
        email: `DonCaron@gmail.com`,
        name: `DonCaron`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`
      }
    },
    authorizationStatus: `AUTH`
  },
  DATA: {
    film: {
      title: `Mind Hunter`,
      genre: `Mystery`,
      releaseDate: 2005,
      poster: `/img/the-grand-budapest-hotel-poster.jpg`,
      preview: `/img/mindhunter.jpg`,
      backgroundPoster: `/img/bg-header.jpg`,
      description: `FBI agent tries to solve murders all around the USA`,
      rating: 10.0,
      ratingAmount: 350,
      director: `Dan Desk`,
      cast: [`Kim Check`, `Alan Rickman`, `Peter Brower`],
      runningTime: 50,
      filmPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      id: 3,
      backgroundColor: `#73B39A`,
      fullVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      isFavorite: false,
    },
    films: [
      {
        title: `Fantastic Beasts`,
        genre: `Fantasy`,
        releaseDate: 2018,
        poster: `/img/the-grand-budapest-hotel-poster.jpg`,
        preview: `/img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        backgroundPoster: `/img/bg-header.jpg`,
        description: `New amazing story from J.K.Rowling, new adventures in wizards world`,
        rating: 10.0,
        ratingAmount: 500,
        director: `Curt Russel`,
        cast: [`Kim Check`, `Alan Rickman`, `Peter Brower`],
        runningTime: 20,
        filmPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        id: 1,
        backgroundColor: `#73B39A`,
        fullVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        isFavorite: false,

      },
      {
        title: `Zombi Jack`,
        genre: `Horror`,
        releaseDate: 2020,
        poster: `/img/the-grand-budapest-hotel-poster.jpg`,
        preview: `/img/midnight-special.jpg`,
        backgroundPoster: `/img/bg-header.jpg`,
        description: `New flash eating virus trying to destory the world`,
        rating: 5.5,
        ratingAmount: 30,
        director: `Wes Kraven`,
        cast: [`Kim Check`, `Alan Rickman`, `Peter Brower`],
        runningTime: 30,
        filmPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        id: 2,
        backgroundColor: `#73B39A`,
        fullVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        isFavorite: false,

      },
      {
        title: `Mind Hunter`,
        genre: `Mystery`,
        releaseDate: 2005,
        poster: `/img/the-grand-budapest-hotel-poster.jpg`,
        preview: `/img/mindhunter.jpg`,
        backgroundPoster: `/img/bg-header.jpg`,
        description: `FBI agent tries to solve murders all around the USA`,
        rating: 10.0,
        ratingAmount: 350,
        director: `Dan Desk`,
        cast: [`Kim Check`, `Alan Rickman`, `Peter Brower`],
        runningTime: 40,
        filmPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        id: 3,
        backgroundColor: `#73B39A`,
        fullVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
        isFavorite: false,

      },
    ]
  }
});

store.dispatch(setPromoFilm(FILM));
export {store};
