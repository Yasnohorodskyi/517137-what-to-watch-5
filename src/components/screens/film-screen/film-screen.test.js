import TestRenderer from "react-test-renderer";
import FilmScreen from "./film-screen";
import {BrowserRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import * as mock from "../../../mock-test";


test(`render Film Screen`, () => {
  const tree = TestRenderer
    .create(
        <Provider store={mock.store}>
          <BrowserRouter>
            <FilmScreen
              film={mock.FILM}
              authorizationStatus={`AUTH`}
              btnPlayHandler={mock.noop}
            />
          </BrowserRouter>
        </Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});
