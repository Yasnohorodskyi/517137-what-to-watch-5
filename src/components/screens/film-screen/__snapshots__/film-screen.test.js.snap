// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Film Screen 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#795433",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Pumping"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <div
            className="logo"
          >
            <a
              className="logo__link "
              href="/"
              onClick={[Function]}
            >
              <span
                className="logo__letter logo__letter--1"
              >
                W
              </span>
              <span
                className="logo__letter logo__letter--2"
              >
                T
              </span>
              <span
                className="logo__letter logo__letter--3"
              >
                W
              </span>
            </a>
          </div>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Pumping
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/films/5/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Pumping"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  data-tab="Overview"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  data-tab="Details"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  data-tab="Reviews"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8.6
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Awesome
              </span>
              <span
                className="movie-rating__count"
              >
                109661
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              A young man who was sentenced to seven years
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Nicolas Winding Refn
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Tom Hardy, Kelly Adams, Luing Andrews
                 and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link logo__link--light"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2020 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
